helm/resource_release.go

```diff
@@ -164,6 +164,7 @@ func resourceRelease() *schema.Resource {
 				ForceNew:    true,
 				Default:     "default",
 				Description: "Namespace to install the release into.",
+				DefaultFunc: schema.EnvDefaultFunc("HELM_NAMESPACE", "default"),
```

Comment:
This env variable seems to be used in helm core as well so I figured this might be the right place for it after removing `helm_namespace` from the provider config.